pytube/__main__.py

```diff
@@ -247,6 +254,21 @@ def vid_info(self):
         self._vid_info = innertube_response
         return self._vid_info
 
+    def update_vid_info_with(self, client='ANDROID'):
+        """Attempt to update the vid_info by using client passed as arg."""
+        innertube = InnerTube(
+            client=client,
+            use_oauth=self.use_oauth,
+            allow_cache=self.allow_oauth_cache
+        )
+
+        innertube_response = innertube.player(self.video_id)
+        playability_status = innertube_response['playabilityStatus'].get('status', None)
+        # If we still can't access the video, raise an exception
+        if playability_status == 'UNPLAYABLE':
+            raise exceptions.VideoUnavailable(self.video_id)
+        self._vid_info = innertube_response
+
```

Comment:
At first glance I would say this entire method should be part of `YouTube.vid_info`, no need for a separate detour like that? `vid_info` could take a default `client` parameter?